src/App/Fossa/Container/Sources/DockerTarball.hs

```diff
@@ -0,0 +1,213 @@
+          ]
+      )
+      imageDigest
+
+analyzeLayer ::
+  ( Has Diagnostics sig m
+  , Has Exec sig m
+  , Has (Lift IO) sig m
+  , Has Logger sig m
+  , Has Telemetry sig m
+  , Has Debug sig m
+  ) =>
+  Int ->
+  OsInfo ->
+  SomeFileTree TarEntryOffset ->
+  Path Abs File ->
+  m [SourceUnit]
+analyzeLayer capabilities osInfo layerFs tarball =
+  map (Srclib.toSourceUnit noUnusedDeps)
+    . mapMaybe toProjectResult
+    <$> (runReader noFilters)
+      ( do
+          (projectResults, ()) <-
+            runTarballReadFSIO layerFs tarball
+              . runReader noExperimental
+              . Diag.context "discovery/analysis tasks"
+              . runOutput @DiscoveredProjectScan
+              . runStickyLogger SevInfo
+              . runFinally
+              . withTaskPool capabilities updateProgress
+              . runAtomicCounter
+              $ do
+                runAnalyzers osInfo noFilters
+          pure projectResults
+      )
+  where
+    noExperimental :: ExperimentalAnalyzeConfig
+    noExperimental = ExperimentalAnalyzeConfig Nothing
+
+    noUnusedDeps :: Bool
+    noUnusedDeps = False
+
+    noFilters :: AllFilters
+    noFilters = mempty
+
+runAnalyzers ::
+  ( AnalyzeTaskEffs sig m
+  , Has (Output DiscoveredProjectScan) sig m
+  , Has TaskPool sig m
+  , Has AtomicCounter sig m
+  ) =>
+  OsInfo ->
+  AllFilters ->
+  m ()
+runAnalyzers osInfo filters = do
+  traverse_
+    single
+    [ DiscoverFunc (Apk.discover osInfo)
+    ]
+  where
+    single (DiscoverFunc f) = withDiscoveredProjects f basedir (runDependencyAnalysis basedir filters)
+    basedir = Path $ toString fixedVfsRoot
+
+runDependencyAnalysis ::
+  ( AnalyzeProject proj
+  , Has (Lift IO) sig m
+  , Has AtomicCounter sig m
+  , Has Debug sig m
+  , Has Logger sig m
+  , Has ReadFS sig m
+  , Has Exec sig m
+  , Has (Output DiscoveredProjectScan) sig m
+  , Has (Reader ExperimentalAnalyzeConfig) sig m
+  , Has (Reader AllFilters) sig m
+  , Has Stack sig m
+  , Has Telemetry sig m
+  ) =>
+  Path Abs Dir ->
+  AllFilters ->
+  DiscoveredProject proj ->
+  m ()
+runDependencyAnalysis basedir filters project@DiscoveredProject{..} = do
+  let dpi = DiscoveredProjectIdentifier projectPath projectType
+  case applyFiltersToProject basedir filters project of
+    Nothing -> do
+      logInfo $ "Skipping " <> pretty projectType <> " project at " <> viaShow projectPath <> ": no filters matched"
+      output $ SkippedDueToProvidedFilter dpi
+    Just targets -> do
+      logInfo $ "Analyzing " <> pretty projectType <> " project at " <> pretty (toFilePath projectPath)
+      let ctxMessage = "Project Analysis: " <> showT projectType
+      graphResult <- Diag.runDiagnosticsIO . diagToDebug . stickyLogStack . withEmptyStack . Diag.context ctxMessage $ do
+        analyzeProject targets projectData
+      Diag.flushLogs SevError SevDebug graphResult
+      output $ Scanned dpi (mkResult basedir project <$> graphResult)
```

Comment:
I notice we explicitly set no filters above. Does the existence of filter handling code imply that filters will be supported later? if so please consider adding todo comments on things that are stubbed vs intentionally unimplemented - if not please consider dropping the filter handling code.